src/ejabberd_service.erl

```diff
@@ -136,10 +136,7 @@ socket_type() ->
 -spec process_packet(Acc :: mongoose_acc:t(), From :: jid:jid(), To :: jid:jid(),
     El :: exml:element(), Pid :: pid()) -> any().
 process_packet(Acc, From, To, El, Pid) ->
-    Pid ! {route, From, To, mongoose_acc:strip(#{ lserver => From#jid.lserver,
```

Comment:
I'm wondering now that it could break some logic (such as metrics? calls to DB pools?) because now the `lserver` is taken from the recipient JID, which totally makes sense but in case of components and PubSub actually changes the value in this acc field (when compared to the original code).

I'm not saying it's wrong but have considered all consequences?

Reply:
tró, I didn't notice that there is a subhost mapping in the meantime